File: readability/readability.c
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

char* get_text(void);
int get_number_of_sentences(char* text);
int get_number_of_letters(char* text);
int get_number_of_words(char* text);

int main(void)
{
    // Count the number of letters, words, and sentences in the text
    char* text = get_text();
    int sentences = get_number_of_sentences(text);
    int letters = get_number_of_letters(text);
    int words = get_number_of_words(text);

    // Compute the Coleman-Liau index
    float coleman_liau_index = 0.0588 * letters/words * 100 - 0.296 * sentences/words * 100 - 15.8;
    if (coleman_liau_index <= 0)
    {
        printf("Before grade 1\n");
    } else if (coleman_liau_index >= 16)
    {
        printf("Grade 16+\n");
    } else
    {
        printf("Grade %d\n", (int) round(coleman_liau_index));
    }   
}

char* get_text(void)
{
    static char text[500]; 
    printf("Enter text: ");
    fgets(text, sizeof(text), stdin);  
    return text;
}


int get_number_of_sentences(char* text)
{
    int count = 0;
    int size = strlen(text);
    for (int i = 0; i < size; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            count++;
        }
    }
return count;
}

int get_number_of_letters(char* text)
{
    int count = 0;
    int size = strlen(text);
    for (int i = 0; i < size; i++)
    {
        if (isalpha(text[i]))  // Check if the character is a letter
        {
            count++;
        }
    }
return count;
}

int get_number_of_words(char* text)
{
    int count = 1;
    int size = strlen(text);
    for (int i = 0; i < size; i++)
    {
        if (text[i] == ' ')  // Check if the character is a space
        {
            count++;
        }
    }
return count;
}

File: projects_w6/readability.py
def main():
    text = input("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    grade = coleman_liau_index(letters, words, sentences)

    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")

def count_letters(text):
    return sum(1 for char in text if char.isalpha())

def count_words(text):
    return len(text.split())

def count_sentences(text):
    return sum(1 for char in text if char in '.!?')

def coleman_liau_index(letters, words, sentences):
    L = (letters / words) * 100
    S = (sentences / words) * 100
    index = 0.0588 * L - 0.296 * S - 15.8
    return round(index)

main()